demo/admin/src/products/ProductVariantForm.tsx

```diff
@@ -0,0 +1,140 @@
+          };
+
+    const saveConflict = useFormSaveConflict({
+        checkConflict: async () => {
+            const updatedAt = await queryUpdatedAt(client, "productVariant", id);
+            return resolveHasSaveConflict(data?.productVariant.updatedAt, updatedAt);
+        },
+        formApiRef,
+        loadLatestVersion: async () => {
+            await refetch();
+        },
+    });
+
+    const handleSubmit = async (formValues: FormValues, form: FormApi<FormValues>, event: FinalFormSubmitEvent) => {
+        if (await saveConflict.checkForConflicts()) throw new Error("Conflicts detected");
+        const output = {
+            ...formValues,
+            image: rootBlocks.image.state2Output(formValues.image),
+        };
+        if (mode === "edit") {
+            if (!id) throw new Error();
+            await client.mutate<GQLUpdateProductVariantMutation, GQLUpdateProductVariantMutationVariables>({
+                mutation: updateProductVariantFormMutation,
+                variables: { id, input: output, lastUpdatedAt: data?.productVariant.updatedAt },
+            });
+        } else {
+            const { data: mutationReponse } = await client.mutate<GQLCreateProductVariantMutation, GQLCreateProductVariantMutationVariables>({
+                mutation: createProductVariantFormMutation,
+                variables: { product: productId, input: output },
+            });
+            if (!event.navigatingBack) {
+                const id = mutationReponse?.createProductVariant.id;
+                if (id) {
+                    setTimeout(() => {
+                        stackSwitchApi.activatePage(`edit`, id);
+                    });
+                }
+            }
+        }
+    };
+
+    if (error) {
+        return <FormattedMessage id="common.error" defaultMessage="An error has occured. Please try again at later" />;
+    }
+
+    if (loading) {
+        return <CircularProgress />;
+    }
+
+    return (
+        <FinalForm<FormValues>
+            apiRef={formApiRef}
+            onSubmit={handleSubmit}
+            mode="edit"
+            initialValues={initialValues}
+            initialValuesEqual={isEqual} //required to compare block data correctly
+            onAfterSubmit={(values, form) => {
+                //don't go back automatically TODO remove this automatismn
+            }}
+            subscription={{}}
+        >
+            {() => (
+                <EditPageLayout>
+                    {saveConflict.dialogs}
+                    <MainContent>
+                        <Field
+                            fullWidth
+                            name="name"
+                            component={FinalFormInput}
+                            label={<FormattedMessage id="productVariant.name" defaultMessage="Name" />}
+                        />
+                        <Field name="image" isEqual={isEqual}>
+                            {createFinalFormBlock(rootBlocks.image)}
```

Comment:
Out of scope: We should memo this. For instance, when a Rich Text Block is created, every render loses the current cursor position